libs/editor/egui_editor/src/layout_job.rs

```diff
@@ -28,127 +30,122 @@ pub enum Annotation {
 }
 
 impl LayoutJobInfo {
-    pub fn new(
```

Comment:
`new`/`append` refactored for clarity:
* `calc()` below checks `block_start` to decide whether to create or append; only use of `block_start`
* only `create` calculates annotation and head size
* all annotations identified in one place (new `annotation_and_head_tail_size` helper fn below)